### PR TITLE
[4.12] Unrevert rebase with ginkgo test init duplication removed

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

File: pkg/test/ginkgo/cmd_runsuite.go [hunk 1]
```diff
@@ -183,22 +182,6 @@ func (opt *Options) Run(suite *TestSuite, junitSuiteName string) error {
 		}
 	}
 
-	// This ensures that tests in the identified paths do not run in parallel, because
-	// the test suite reuses shared resources without considering whether another test
-	// could be running at the same time. While these are technically [Serial], ginkgo
-	// parallel mode provides this guarantee. Doing this for all suites would be too
-	// slow.
-	setTestExclusion(tests, func(suitePath string, t *testCase) bool {
```
Comment (deads2k) [inline pkg/test/ginkgo/cmd_runsuite.go:191]: where did this move and if straight removed, how did you determine this was now safe?

Comment (bparees) [reply pkg/test/ginkgo/cmd_runsuite.go:191]: @soltysh ?  From what I can tell I didn't have this dropped in my ginkgo v2 rebase

**Comment by kikisdeliveryservice** [comment]
seems to have an issue on verify-deps: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27507/pull-ci-openshift-origin-release-4.12-verify-deps/1587576646750179328

/payload 4.12 nightly blocking

**Comment by bparees** [comment]
> seems to have an issue on verify-deps: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27507/pull-ci-openshift-origin-release-4.12-verify-deps/1587576646750179328

the deps issue is because this PR is carrying a patch to ginkgo.  The commit really needs to go into openshift/onsi-gingko and then get vendored here.

I assume we'll do that once we're confident of all the changes we need.

**Comment by bparees** [comment]
being pursued in https://github.com/openshift/origin/pull/27526 (4.13, but will be backported).

**Pull request closed without merging**